django/db/models/sql/compiler.py

```diff
@@ -551,11 +551,14 @@ def as_sql(self, with_limits=True, with_col_aliases=False):
                     if self._meta_ordering:
                         # When the deprecation ends, replace with:
                         # order_by = None
+                        ordering = "', '".join([str(f) for f in self._meta_ordering])
+                        ordering = ordering.replace("'OrderBy'", "OrderBy").replace(")'", ")")
+                        ordering = re.sub(r"F\((\w+)\)", r"F('\1')", ordering)
                         warnings.warn(
                             "%s QuerySet won't use Meta.ordering in Django 3.1. "
                             "Add .order_by('%s') to retain the current query." % (
                                 self.query.model.__name__,
-                                "', '".join(self._meta_ordering)
+                                ordering
```

Comment:
`OrderBy` expression can be more complicated e.g. it can contain different kind of expressions e.g. `Value()`, `Case()` etc. We will not be able to predict all use cases with regexp. Moreover IMO we don't need to provide message that can be copy & paste directly into queryset. Using `repr()` should be enough for clear message, e.g.
```diff
diff --git a/django/db/models/sql/compiler.py b/django/db/models/sql/compiler.py
index 23bc339e78..83c10183a3 100644
--- a/django/db/models/sql/compiler.py
+++ b/django/db/models/sql/compiler.py
@@ -553,9 +553,9 @@ class SQLCompiler:
                         # order_by = None
                         warnings.warn(
                             "%s QuerySet won't use Meta.ordering in Django 3.1. "
-                            "Add .order_by('%s') to retain the current query." % (
+                            "Add .order_by(%s) to retain the current query." % (
                                 self.query.model.__name__,
-                                "', '".join(self._meta_ordering)
+                                ', '.join(repr(f) for f in self._meta_ordering),
                             ),
                             RemovedInDjango31Warning,
                             stacklevel=4,
```
```